### PR TITLE
Cleaning up Old Mixins

### PR DESCRIPTION
**Pull request by TechTastic**
I cleaned up the code within `MixinTurtleBrain` and `MixinTurtleMoveCommand` as well as updated the CC: Tweaked and CC: Restitched dependencies on both Forge and Fabric

### CODE REVIEW

**Comment by TechTastic** [comment]
Ok, it uses the new API now, also added missing `@Pseudo` to `MixinSpeakerSound` on both Forge and Fabric

File: fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java [hunk 1]
```diff
@@ -42,16 +42,18 @@ public abstract class MixinTurtleMoveCommand {
...
+            ValkyrienSkies.positionToNearbyShipsAndWorld(world, position.getX() + 0.5, position.getY() + 0.5,
                 position.getZ() + 0.5, 0.1);
         for (final Vector3d vec : detectedShips) {
             if (vec != null) {
                 return vec;
             }
         }
-        return new Vector3d(position.getX(), position.getY(), position.getZ());
+
+        return VectorConversionsMCKt.toJOMLD(position);
```
Comment (Rubydesic) [inline fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java:57]: This should be `ValkyrienSkies.toJOMLd`

File: fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java [hunk 2]
```diff
@@ -42,16 +42,18 @@ public abstract class MixinTurtleMoveCommand {
     }
 
     //CUSTOM METHODS
+
     @Unique
     private static Vector3d getShipPosFromWorldPos(final Level world, final BlockPos position) {
```
Comment (Rubydesic) [inline fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java:47]: This logic is broken when multiple ships are intersecting the position. Can you implement the fix from https://github.com/ValkyrienSkies/Valkyrien-Skies-2/pull/1035?

File: fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleBrain.java [hunk 3]
```diff
@@ -45,71 +39,27 @@ public abstract class MixinTurtleBrain {
...
         final Level world = getLevel();
 
-        final Ship ship = VSGameUtilsKt.getShipManagingPos(world, oldPos);
+        final Ship ship = ValkyrienSkies.getShipManagingBlock(world, oldPos);
         if (ship != null) {
             // THERE IS A SHIP
-            final Direction d = getNewDirection(ship, currentOwner.getDirection());
-            if (!doesShipContainPoint(ship, pos)) {
+            final Vector3d transformedDirection = ship.getShipToWorld().transformDirection(
+                VectorConversionsMCKt.toJOMLD(currentOwner.getDirection().getNormal())
```
Comment (Rubydesic) [inline fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleBrain.java:46]: `ValkyrienSkies.toJOMLd`

File: forge/src/main/java/org/valkyrienskies/mod/forge/mixin/compat/cc_tweaked/MixinTurtleMoveCommand.java [hunk 4]
```diff
@@ -42,16 +42,18 @@ public abstract class MixinTurtleMoveCommand {
...
+            ValkyrienSkies.positionToNearbyShipsAndWorld(world, position.getX() + 0.5, position.getY() + 0.5,
                 position.getZ() + 0.5, 0.1);
         for (final Vector3d vec : detectedShips) {
             if (vec != null) {
                 return vec;
             }
         }
-        return new Vector3d(position.getX(), position.getY(), position.getZ());
+
+        return VectorConversionsMCKt.toJOMLD(position);
```
Comment (Rubydesic) [inline forge/src/main/java/org/valkyrienskies/mod/forge/mixin/compat/cc_tweaked/MixinTurtleMoveCommand.java:57]: `ValkyrienSkies.toJOMLd`

**Review by Rubydesic — CHANGES_REQUESTED** [review]

File: fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java (hunk 1 above)
Comment (TechTastic) [reply fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java:57]: Ah, so thats where the method moved to

Comment (TechTastic) [reply fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java]: Fixed

File: fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java (hunk 2 above)
Comment (TechTastic) [reply fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleMoveCommand.java:58]: Implemented

File: fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleBrain.java (hunk 3 above)
Comment (TechTastic) [reply fabric/src/main/java/org/valkyrienskies/mod/fabric/mixin/compat/cc_restitched/MixinTurtleBrain.java]: Fixed

File: forge/src/main/java/org/valkyrienskies/mod/forge/mixin/compat/cc_tweaked/MixinTurtleMoveCommand.java (hunk 4 above)
Comment (TechTastic) [reply forge/src/main/java/org/valkyrienskies/mod/forge/mixin/compat/cc_tweaked/MixinTurtleMoveCommand.java]: Fixed

**Review by Rubydesic — APPROVED** [review]

**Pull request merged**